openlibrary/catalog/marc/parse.py

```diff
@@ -384,7 +384,7 @@ def name_from_list(name_parts: list[str]) -> str:
     return remove_trailing_dot(name)
 
 
-def read_author_person(f, tag='100') -> dict:
+def read_author_person(field, tag: str = '100') -> dict | None:
```

Comment:
Is `field` a `BinaryDataField` or a `DataField`?
```suggestion**
def read_author_person(field: BinaryDataField, tag: str = '100') -> dict | None:
```

Reply:
I honestly don't know -- it looks like it _should_ be either, but in practice the import is only used with binary MARCs, and I don't have much confidence it will work with individual MARC XML inputs.

Theoretically it should though.

I'm not sure parse.py should even know about `BinaryDataField` or `DataField`s, and the fact is needs to, and that they don't even share a type makes me think `MarcBinary`, `MarcXml`, and `MarcBase` need an overhaul -- which is a bigger thread than I wanted to tug at to add this feature. :(

None of the existing MARC XML tests fail with this new code, but that just makes me suspect they don't have sufficient coverage.

Bringing `MarcXml` up to the same standard is probably required. Either that, or deprecating that import format until we have a use for it and are convinced it works to the same level as the binary import path.

Reply:
@cclauss Next step for this feature is to add the 880 import behavior to `MarcXml`, using at least the existing test data (at least one has Hebrew which will work for testing this feature) -- I'll figure out the correct type ranges with that. 

Stay tuned.

Thanks for your reviews so far. This has been getting a bit complicated. While I feel like I'm often fighting with mypy, the type hints do help understand what some of these old methods are trying to do.

Reply:
Resolved by creating a `MarcFieldBase` base class to cover both cases.